crypto/s2n_cipher.h

```diff
@@ -82,7 +82,7 @@ struct s2n_cipher {
         struct s2n_composite_cipher comp;
     } io;
     uint8_t key_material_size;
-    uint8_t (*is_available)(void);
+    S2N_RESULT (*is_available)(bool *available);
     int (*init)(struct s2n_session_key *key);
     int (*set_decryption_key)(struct s2n_session_key *key, struct s2n_blob *in);
     int (*set_encryption_key)(struct s2n_session_key *key, struct s2n_blob *in);
```

Comment:
So what the linked issue meant by "While we're fixing that, might as well make all the methods return S2N_RESULT" was to also update all these methods from int to S2N_RESULT. Do you plan to do that in a separate PR? Resolving this PR will currently resolve that issue, since you used the "Resolves" tag in your description.

Reply:
The size of the change might be large if all is combined, so I will remove the "Resolves" tag and address the rest in separate PRs.